qcore/constants.py

```diff
@@ -105,6 +105,58 @@ def iterate_str_values(cls, ignore_none=True):
             yield item.str_value
 
 
+class DependencyTarget(Enum):
+    MEDIAN = auto()
+    REL = auto()
+
+
+class Dependency:
+    """
+    A ProcessType, DependencyTarget pair.
+    With just in time generation to prevent circular declarations with the ProcessType enum
+    """
+
+    _process: "ProcessType"
+    _dependency: DependencyTarget
+
+    def __init__(
+        self,
+        proc_id: Union[int, "ProcessType"],
+        median: Union[bool, DependencyTarget] = False,
+    ):
+        self._process = proc_id
+        self._dependency = median
+
+    @property
+    def process(self):
+        if isinstance(self._process, int):
+            self._process = ProcessType(self._process)
+        elif isinstance(self._process, tuple):
+            self._process = ProcessType(self._process[0])
+        return self._process
+
+    @property
+    def dependency(self):
+        if self._dependency is True:
+            self._dependency = DependencyTarget.MEDIAN
+        elif self._dependency is False:
+            self._dependency = DependencyTarget.REL
```

Comment:
Probably don't need the extra Boolean for dependency, can just be a default to DependencyTarget.REL